### PR TITLE
Alisha.zariff

### PR DESCRIPTION
**Pull request by nadaalomranii**

### CODE REVIEW

**Review by Sahirara — CHANGES_REQUESTED** [review]
Overall looks really good. Some main issues were occuring:
- CommonAssignment entity: the class cannot have update grade (i.e. a setter) since it violates CA
- AddAssignmentCourseDataAccessInterface doesn't implement DataAccessInetrface: make sure it does
- At this current moment, not sure what conflict there is with AssignmentListView. Once you make the previous fixes try running this file using AppBuilder to see if there are any problems

File: src/main/java/interface_adapter/edit_assignment/EditAssignmentPresenter.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+        // since the output data only contains the new score
+        // We still fire the property changed event, but just to let the view know that
+        // it can alert the user that their assignment was changed successfully..
+        editAssignmentViewModel.firePropertyChanged("Assignment Edited");
+
+    }
+
+    @Override
+    public void prepareFailView(String error) {
+        // note: this use case currently can't fail
```
Comment (miralyousef) [inline src/main/java/interface_adapter/edit_assignment/EditAssignmentPresenter.java:29]: what about when the new assignment name is the same as a dif assignment name? does that not need a failview?

**Review by miralyousef — APPROVED** [review]
Your code follows clean architecture and looks great!

Some steps moving forward:
- make sure you implement prepareFailView 
- make sure you have the appropriate documentation 

You're killing it!!

**Review by Sahirara — APPROVED** [review]
Looks good! Great job :)

**Pull request merged**